File: app/src/main/java/com/fishweeb/practical/GunInfo.java
package com.fishweeb.practical;

enum SHOOTTYPE
{
    S_STRAIGHT,
    S_SPREAD;
}

public class GunInfo
{
    private int WeaponDamage;
    private float FireRate;
    private float FireRatePercentage = 100;
    private int ShootAmount;
    private int BonusShootAmount = 0;
    private SHOOTTYPE shoottype;

    public void SetWeaponDamage(int _damage)
    {
        WeaponDamage = _damage;
    }

    public int GetWeaponDamage()
    {
        return WeaponDamage;
    }

    public void SetFireRate(float _firerate)
    {
        FireRate = _firerate;
    }

    public float GetFireRate()
    {
        return FireRate * (FireRatePercentage/100);
    }

    public void SetShootType(SHOOTTYPE _type)
    {
        shoottype = _type;
    }

    public SHOOTTYPE GetShootType()
    {
        return shoottype;
    }

    public void SetShootAmount(int _amount)
    {
        ShootAmount = _amount;
    }

    public void AddShootAmount(SHOOTTYPE _type)
    {
        if (this.shoottype == _type)
            ++BonusShootAmount;
        else
        {
            this.shoottype = _type;
            BonusShootAmount = 1;
        }
    }

    public void SetBonusShootAmount(int _value)
    {
        BonusShootAmount = _value;
    }

    public void SetFireRatePercentage(float _value)
    {
        FireRatePercentage = _value;
    }

    public int GetShootAmount()
    {
        return ShootAmount + BonusShootAmount;
    }
}

File: app/src/main/java/edu/androidgroupproject/GunInfo.kt
package com.fishweeb.practical

enum class SHOOTTYPE {
    S_STRAIGHT,
    S_SPREAD
}

class GunInfo {
    private var WeaponDamage = 0
    private var FireRate = 0f
    private var FireRatePercentage = 100f
    private var ShootAmount = 0
    private var BonusShootAmount = 0
    private var shoottype: SHOOTTYPE? = null

    fun SetWeaponDamage(_damage: Int) {
        WeaponDamage = _damage
    }

    fun GetWeaponDamage(): Int {
        return WeaponDamage
    }

    fun SetFireRate(_firerate: Float) {
        FireRate = _firerate
    }

    fun GetFireRate(): Float {
        return FireRate * (FireRatePercentage / 100)
    }

    fun SetShootType(_type: SHOOTTYPE?) {
        shoottype = _type
    }

    fun GetShootType(): SHOOTTYPE? {
        return shoottype
    }

    fun SetShootAmount(_amount: Int) {
        ShootAmount = _amount
    }

    fun AddShootAmount(_type: SHOOTTYPE) {
        if (this.shoottype == _type) ++BonusShootAmount
        else {
            this.shoottype = _type
            BonusShootAmount = 1
        }
    }

    fun SetBonusShootAmount(_value: Int) {
        BonusShootAmount = _value
    }

    fun SetFireRatePercentage(_value: Float) {
        FireRatePercentage = _value
    }

    fun GetShootAmount(): Int {
        return ShootAmount + BonusShootAmount
    }
}
